server/trillian_log_signer/main.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/extension/builtin"
+	"github.com/google/trillian/server"
+	"github.com/google/trillian/util"
+	"golang.org/x/net/context"
+)
+
+var (
+	exportRPCMetrics              = flag.Bool("export_metrics", true, "If true starts HTTP server and exports stats")
+	httpPortFlag                  = flag.Int("http_port", 8091, "Port to serve HTTP metrics on")
+	sequencerSleepBetweenRunsFlag = flag.Duration("sequencer_sleep_between_runs", time.Second*10, "Time to pause after each sequencing pass through all logs")
+	batchSizeFlag                 = flag.Int("batch_size", 50, "Max number of leaves to process per batch")
+	numSeqFlag                    = flag.Int("num_sequencers", 10, "Number of sequencers to run in parallel")
+	sequencerGuardWindowFlag      = flag.Duration("sequencer_guard_window", 0, "If set, the time elapsed before submitted leaves are eligible for sequencing")
+)
+
+func startHTTPServer(port int) error {
+	sock, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+	if err != nil {
+		return err
+	}
+	go func() {
+		glog.Info("HTTP server starting")
+		http.Serve(sock, nil)
+	}()
+
+	return nil
+}
+
+func awaitSignal(cancel context.CancelFunc) {
```

Comment:
Could you add a TODO to share this function between servers? I suppose it's the same everywhere.